CA1861 Analyzer and code fix documentation: AvoidConstArrays (#25659)
dotnet/docs: docs/fundamentals/code-analysis/quality-rules/ca1861.md

---
title: "CA1861: Avoid constant arrays as arguments (code analysis)"
description: "Learn about code analysis rule CA1861: Avoid constant arrays as arguments"
ms.date: 05/10/2023
ms.topic: reference
f1_keywords:
  - "AvoidConstArrays"
  - "CA1861"
helpviewer_keywords:
  - "AvoidConstArrays"
  - "CA1861"
author: steveberdy
dev_langs:
  - CSharp
  - VB
---

# CA1861: Avoid constant arrays as arguments

|                                     | Value                                  |
| ----------------------------------- | -------------------------------------- |
| **Rule ID**                         | CA1861                                 |
| **Category**                        | [Performance](performance-warnings.md) |
| **Fix is breaking or non-breaking** | Non-breaking                           |
| **Enabled by default in .NET 7**    | No                                     |

## Cause

A constant array, of literal values, passed to a method via regular invocation or extension method invocation.

## Rule description

> Constant arrays passed as arguments are not reused which implies a performance overhead. Consider extracting them to 'static readonly' fields to improve performance.

Reviewer: ```suggestion
Constant arrays passed as arguments aren't reused, which implies a performance overhead. Consider extracting them to `static readonly` fields to improve performance.
```

> Constant arrays passed as arguments are not reused which implies a performance overhead. Consider extracting them to 'static readonly' fields to improve performance.

Reviewer: ```suggestion
Constant arrays passed as arguments aren't reused, which implies a performance overhead. Consider extracting them to `static readonly` fields to improve performance.
```
